test/serializers/associations_test.rb

```diff
@@ -261,6 +261,36 @@ def false
 
           assert_equal(expected, hash)
         end
+
+        def test_string_conditional_associations
+          serializer = Class.new(ActiveModel::Serializer) do
+            belongs_to :if_assoc_included, if: 'object.true'
+            belongs_to :if_assoc_excluded, if: 'object.false'
+            belongs_to :unless_assoc_included, unless: 'object.false'
+            belongs_to :unless_assoc_excluded, unless: 'object.true'
+          end
+
+          model = ::Model.new(true: true, false: false)
+          hash = serializable(model, serializer: serializer).serializable_hash
+          expected = { if_assoc_included: nil, unless_assoc_included: nil }
+
+          assert_equal(expected, hash)
+        end
+
+        def test_lambda_conditional_associations
+          serializer = Class.new(ActiveModel::Serializer) do
+            belongs_to :if_assoc_included, if: -> (s) { s.object.true }
```

Comment:
Isn't the rails way these are used to evaluate in the context of the instance? i.e. `-> { object.true }`?